src/utils/Common.ts

```diff
@@ -0,0 +1,27 @@
+type Mappable = {
+  [key: string]: any;
+};
+
+export const getPropertyFromPath = (object: Mappable, path: string | string[]): any => {
```

Comment:
This is quite a condensed method. I think it'd help to both rename iterators from `i` and to add some comments explaining why it's structured like this.

Reply:
Another thing is that this method could likely be handled slightly nicer recursively by doing something like:
- Check if first index of path exists
  - If the path doesn't exist, the property is unnavigable. Return undefined.
  - If it does, recurse into the property and pass the new path (original being path[0..n], new being path[1..n])
- Once `path` is empty, return the object.